pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -65,15 +67,57 @@ func (a *index) ServicesCollection(
 				multicluster.ClusterKRTMetadataKey: clusterID,
 			}),
 		)...)
+
 	ServiceEntriesInfo := krt.NewManyCollection(serviceEntries, a.serviceEntryServiceBuilder(waypoints, namespaces),
 		append(
 			opts.WithName("ServiceEntriesInfo"),
 			krt.WithMetadata(krt.Metadata{
 				multicluster.ClusterKRTMetadataKey: clusterID,
 			}),
 		)...)
-	WorkloadServices := krt.JoinCollection(
-		[]krt.Collection[model.ServiceInfo]{ServicesInfo, ServiceEntriesInfo},
+	serviceEntryByHostname := krt.NewIndex(ServiceEntriesInfo, "serviceEntryByHostname", func(se ServiceEntryInfo) []string {
+		return []string{types.NamespacedName{
+			Name:      se.Service.Hostname,
+			Namespace: "default",
```

Comment:
Why always default namespace?
```suggestion
			Namespace: se.Service.Namespace,
```

Reply:
I really do not know why I would do that :upside_down_face: